datalad/metadata/search.py

```diff
@@ -96,7 +103,54 @@ class Search(Interface):
     @datasetmethod(name='search')
     def __call__(match, dataset, report=None, report_matched=False, format='custom', regex=False):
 
-        ds = require_dataset(dataset, check_installed=True, purpose='dataset search')
+        lgr.debug("Initiating search for match=%r and dataset %r",
+                  match, dataset)
+        try:
+            ds = require_dataset(dataset, check_installed=True, purpose='dataset search')
+        except NoDatasetArgumentFound:
+            exc_info = sys.exc_info()
+            if dataset is None:
+                # none was provided so we could ask user either he possibly wants
+                # to install our beautiful mega-duper-super-dataset?
+                # TODO: following logic could possibly benefit other actions.
+                if exists(LOCAL_CENTRAL_PATH):
+                    central_ds = Dataset(LOCAL_CENTRAL_PATH)
+                    if central_ds.is_installed():
+                        if ui.yesno(
+                            title="No DataLad dataset found at current location",
+                            text="Would you like to search within DataLad "
+                                 "meta-dataset under % r and search within it?"
+                                  % LOCAL_CENTRAL_PATH):
+                            pass
+                    else:
+                        raise NoDatasetArgumentFound(
+                            "No DataLad dataset found at current location and "
+                            "%r already exists but does not contain installed "
+                            "dataset." % LOCAL_CENTRAL_PATH)
+                elif ui.yesno(
+                       title="No DataLad dataset found at current location",
+                       text="Would you like to install stock DataLad "
+                            "meta-dataset under %r?"
+                            % LOCAL_CENTRAL_PATH
+                       ):
+                    from datalad.api import install
+                    central_ds = install(LOCAL_CENTRAL_PATH, source='///')
+                else:
+                    reraise(*exc_info)
+
+                ui.message(
```

Comment:
ATM this should be `lgr.info` instead of `ui.message`, I guess. We should come up with a complete solution but let's be consistent even if it is to be changed.